darts/models/torch_forecasting_model.py

```diff
@@ -616,18 +616,43 @@ def predict_from_dataset(self,
 
                 # repeat prediction procedure for every needed sample
                 batch_predictions = []
-                for i in range(num_samples):
+
+                # if batch_size is not a multiple of input_series, this won't fill the entire batch_size
+                num_series = input_series.shape[0]
+                batch_sample_size = min(max(batch_size // num_series, 1), num_samples)
+                sample_length = 0
+
+                in_shape = input_series.shape
+
+                while sample_length < num_samples:
+                    if sample_length+batch_sample_size > num_samples:
+                        batch_sample_size = num_samples - sample_length
+
+                    # generate prediction for num_series x batch_sample_size
+                    sample_input_series = input_series.tile((batch_sample_size, 1, 1))
+
                     if self.is_recurrent:
-                        batch_prediction = self._predict_batch_recurrent_model(n, input_series, cov_future)
+                        batch_prediction = self._predict_batch_recurrent_model(n, sample_input_series, cov_future)
                     else:
-                        batch_prediction = self._predict_batch_block_model(n, input_series, cov_future, roll_size)
+                        batch_prediction = self._predict_batch_block_model(n, sample_input_series, cov_future, roll_size)
 
-                    # bring predictions into desired format and drop unnecessary values
+                    # concatenate array of predictions into contiguous prediction
                     batch_prediction = torch.cat(batch_prediction, dim=1)
+                    out_shape = batch_prediction.shape
+
+                    # drop unnecessary values
                     batch_prediction = batch_prediction[:, :n, :]
-                    batch_prediction = batch_prediction.cpu().detach().numpy()
+
+                    # reshape from 3d tensor (num_series x batch_sample_size, ...)
+                    # into 4d tensor (batch_sample_size, num_series, ...), where dim 0 represents the samples
+                    batch_prediction = batch_prediction.reshape((batch_sample_size, num_series,) + out_shape[1:])
```

Comment:
This line is causing some issues now it seems, looking at the tests. I see that you changed the order of the statements in the last commit. Any idea about how to fix this?